index.js

```diff
@@ -111,22 +111,37 @@ function toSpriteUrl(url, path, extension) {
 export function applyStyle(layer, glStyle, source, path, resolutions) {
   return new Promise(function(resolve, reject) {
 
+    // TODO: figure out where best place to check source type is
+    // Note that the source arg is an array of gl layer ids and each must be
+    // dereferenced to get source type to validate
     if (typeof glStyle != 'object') {
       glStyle = JSON.parse(glStyle);
     }
     if (glStyle.version != 8) {
-      reject(new Error('glStyle version 8 required.'));
+      return reject(new Error('glStyle version 8 required.'));
+    }
+    if (!(layer instanceof VectorLayer) && !(layer instanceof VectorTileLayer)) {
+      return reject(new Error('Can only apply to VectorLayer or VectorTileLayer'));
+    }
+    const layerSource = layer.getSource();
+    if (!(layerSource instanceof VectorTileSource) && (!layerSource instanceof VectorSource)) {
+      return reject(new Error('Can only apply to VectorTileSource or VectorSource'));
     }
 
     var spriteScale, spriteData, spriteImageUrl, style;
     function onChange() {
       if (!style && (!glStyle.sprite || spriteData)) {
         if (layer instanceof VectorLayer || layer instanceof VectorTileLayer) {
```

Comment:
It still needs to be there because int the documentation it says`applyStyle()` can also be called directly, no?

Reply:
You already added a `reject` in `applyStyle()` when called with a different kind of layer.

For clarification: I meant you can remove the instanceof check from `onChange()`.

Reply:
The problem is in the codepath that creates layers for the user.

`processStyle()` is called from `apply()`, and `processStyle()` calls `finalizeLayer()` on _all_ layers the package creates, including geoJSON and raster layers that may have been created.  The check is done because we don't want `applyStyle()` to fail on layers we create for the user.

If this check is not done, it will cause the promise returned from `applyStyle()` in `onChange()` to reject.  I.e. Layers created by the package itself will cause rejection if this check is taken away.  This means that a mapbox style with anything but vector layers can't be passed to `apply()` without the package erroring out.

The check doesn't have to be here.  We could do a check in the monster for loop to not call `finalizeLayer()` except on vector layers.  Would that be better?

Reply:
Thanks for the explanation. `finalizeLayer()` needs to be called for all layers to make them visible. What about changing things so inside `finalizeLayer()`, `applyStyle()` is only called for `Vector` and `VectorTile` layers?

Reply:
I've added a commit that removes the redundant check.